tools/downloader/topologies.py

```diff
@@ -0,0 +1,75 @@
+import os
+import argparse
+import sys
+
+from pathlib import Path
+from argparse import Namespace
+
+import common
+import downloader
+import converter
+import info_dumper
+
+_cache = Path(os.getenv('OPEN_MODEL_ZOO_CACHE_DIR', Path('.').resolve() / 'models'))
+
+def __load(model_name):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--name', metavar='PAT[,PAT...]',
+        help='only dump info for models whose names match at least one of the specified patterns')
+    parser.add_argument('--list', type=Path, metavar='FILE.LST',
+        help='only dump info for models whose names match at least one of the patterns in the specified file')
+    parser.add_argument('--all', action='store_true', help='dump info for all available models')
+    parser.add_argument('--print_all', action='store_true', help='print all available models')
+    args = Namespace(all=False, list=None, name='mobilenet-ssd', print_all=False)   
+            
+    return common.load_models_from_args(parser, args)[0]
+
+def load_model(model_name, precision='FP32'):
+    download_dir = _cache / __load(model_name).subdirectory
+
+    downloader.main(['--name=' + model_name, '--precisions=' + precision,
+                '--output_dir=' + str(_cache).replace('\\', '\\\\')])
+        
+    return Topology(model_name, download_dir, precision)
+
+def help(model_name):
+    _topology = __load(model_name)
+    description_res = 'Model description for ' + model_name +' \n  ' + \
+                _topology.description + '\n     License: ' +  _topology.license_url
+
+    print(description_res)
+
+class Topology:
+    def __init__(self, name, download_dir, precision):
+        self.name = name
+        self.download_dir = download_dir
+        self.precision = precision
+   
+    def get_ir(self, precision=None):
+        '''
+        Creates OpenVINO Intermediate Representation (IR) network model.
+        If there is already converted model or the model has been downloaded in
+        IR - just returns the paths to the files.
+        Parameters
+        ----------
+        precision
+            Target model precision. By default, the precision from a constructor is used.
+        Returns
+        -------
+        xmlPath
+            Path to generated or downloaded .xml file
+        binPath
+            Path to generated or downloaded .bin file
+        '''
+        precision = precision if precision else self.precision
+        prefix = Path(self.download_dir) / precision / self.name
+
+        xmlPath = str(prefix) + '.xml'
+        binPath = str(prefix) + '.bin'
+        if os.path.exists(xmlPath) and os.path.exists(binPath):
+            return xmlPath, binPath
+
+        converter.main(['--name=' + self.name, '--precisions=' + precision,
+                        '--download_dir=' + str(_cache).replace('\\', '\\\\')])
```

Comment:
Is there `download_dir` for conversion? I thought there is only `output_dir`

Reply:
![image](https://user-images.githubusercontent.com/32199725/88290043-d37a7980-ccfe-11ea-8704-68ecc4c94fa3.png)

Reply:
I see, thanks!